app/policies/board_policy.rb

```diff
@@ -0,0 +1,33 @@
+class BoardPolicy
+  attr_reader :current_user, :board
+
+  def initialize(current_user, board)
+    @current_user = current_user
+    @board = board
+  end
+
+  def index?
+    @current_user.admin? || @current_user.boards.include?(@board)
+  end
+
+  def edit?
+    @current_user.admin?
```

Comment:
Moderatorzy nie mogą zmieniać nic? Plus - skoro to wszystko jest to samo, lepiej to zrobić w osobnej metodzie, łatwiej będzie zmienić przy zmianie założeń co do edycji.